Sources/NotificationApi_Ios_Sdk/AppDelegate/NotificationApiAppDelegate.swift

```diff
@@ -0,0 +1,53 @@
+import UIKit
+
+@available(iOSApplicationExtension, unavailable)
+open class NotificationApiAppDelegate: NSObject, UIApplicationDelegate, UNUserNotificationCenterDelegate {
+    // MARK: - Init
+
+    override init() {
+        super.init()
+
+        UNUserNotificationCenter.current().delegate = self
+        UIApplication.shared.registerForRemoteNotifications()
+    }
+
+    // MARK: - Delegate Functions
+
+    public func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
```

Comment:
This function intercepts the apn token generation and syncs it to our backend. It also calls a hook so developers can act on token generation as well.